src/parser/RdfEscaping.cpp

```diff
@@ -204,6 +215,27 @@ std::string normalizeRDFLiteral(const std::string_view origLiteral) {
   return res;
 }
 
+// ____________________________________________________________________________
+std::string validRDFLiteralFromNormalized(std::string_view normLiteral) {
+  AD_CHECK(normLiteral.starts_with('"'));
```

Comment:
This is what I find odd. Conceptually the step that adds this initial quote in the first place should also be responsible to escape potential inner quotes. Otherwise the generated RDF literal is invalid (if it contains a quote) up until the moment this function is called, which makes this API brittle, i.e. it can be used in a wrong way without any way for the compiler to enforce correct use.